### PR TITLE
use server-side printing in `kubectl get -w`

### LINKED ISSUE #66538: CRD additionalPrinterColumns does not work with kubectl get SOMETHING -w
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:

Without -w option, I can see that kubectl get SOMETHING shows additional columns for my CRD instances.  But with -w flag, just NAME and AGE are shown.

**What you expected to happen**:

additionalPrinterColumns should work with -w flag as well.

**How to reproduce it (as minimally and precisely as possible)**:

1. Define a CRD with additionalPrinterColumns and create its instance.
2. Get it using kubectl get SOMETHING -w

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T20:17:28Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:43:26Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Requests and handles server-side print events for `kubectl get -w`

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubernetes/issues/66538

**Does this PR introduce a user-facing change?**:
```release-note
`kubectl get -w` now prints custom resource definitions with custom print columns
```

/sig cli
/cc @smarterclayton @seans3

### CODE REVIEW

File: pkg/kubectl/cmd/get/get.go [hunk 1]
```diff
@@ -665,8 +673,8 @@ func (o *GetOptions) watch(f cmdutil.Factory, cmd *cobra.Command, args []string)
 			objsToPrint = append(objsToPrint, obj)
 		}
 		for _, objToPrint := range objsToPrint {
-			if o.IsHumanReadablePrinter {
-				// printing always takes the internal version, but the watch event uses externals
+			if o.IsHumanReadablePrinter && objToPrint.GetObjectKind().GroupVersionKind().GroupKind() != tableGK {
+				// printing anything other than tables always takes the internal version, but the watch event uses externals
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/get/get.go]: reminds me for v1 we'll need two versions of this, beta1 and v1

**Comment by smarterclayton** [comment]
looks reasonable, didn't see anything objectionable so far

File: pkg/kubectl/cmd/get/get.go (hunk 1 above)
Comment (liggitt) [reply pkg/kubectl/cmd/get/get.go]: this one in particular just checks group/kind, so it's version neutral... places where we check for *metav1beta1.Table will definitely need to add a switch

**Pull request merged**